### PR TITLE
Fix for JSON serialization exception

### PR DESCRIPTION
**Pull request by lauren-mills**
I found an issue in 4.6 where objects that map to json types (Int64, DateTime) cannot be cast as JTokens in the ObjectPath class. 

Changing to using JToken.FromObject resolves the issue for me, but I'm not sure if there is anything else that needs to be considered with this change.

Error:
```
Test Name:	Test_CalendarCreate_ConfirmNo_ChangeContent
Test FullName:	CalendarSkill.Test.Flow.CreateCalendarFlowTests.Test_CalendarCreate_ConfirmNo_ChangeContent
Test Source:	C:\Users\lamil\source\repos\botframework-solutions\skills\csharp\tests\calendarskill.tests\Flow\CreateCalendarFlowTests.cs : line 272
Test Outcome:	Failed
Test Duration:	0:00:00.2303879

Result StackTrace:	
at Microsoft.Bot.Builder.Dialogs.ObjectPath.GetNormalizedValue(Object value, Boolean json) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\ObjectPath.cs:line 458
   at Microsoft.Bot.Builder.Dialogs.ObjectPath.SetObjectProperty(Object obj, String property, Object value, Boolean json) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\ObjectPath.cs:line 391
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid4[T0,T1,T2,T3](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
   at Microsoft.Bot.Builder.Dialogs.ObjectPath.SetPathValue(Object obj, String path, Object value, Boolean json) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\ObjectPath.cs:line 158
   at Microsoft.Bot.Builder.Dialogs.DialogContext.EndActiveDialogAsync(DialogReason reason, Object result, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\DialogContext.cs:line 563
   at Microsoft.Bot.Builder.Dialogs.DialogContext.EndDialogAsync(Object result, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\DialogContext.cs:line 284
   at Microsoft.Bot.Builder.Dialogs.Prompt`1.ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\Prompts\Prompt.cs:line 157
   at Microsoft.Bot.Builder.Dialogs.DialogContext.ContinueDialogAsync(CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\DialogContext.cs:line 240
   at CalendarSkill.Dialogs.CalendarSkillDialogBase.OnContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken) in C:\Users\lamil\source\repos\botframework-solutions\skills\csharp\calendarskill\Dialogs\CalendarSkillDialogBase.cs:line 100
   at Microsoft.Bot.Builder.Dialogs.ComponentDialog.ContinueDialogAsync(DialogContext outerDc, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\ComponentDialog.cs:line 119
   at Microsoft.Bot.Builder.Dialogs.DialogContext.ContinueDialogAsync(CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\DialogContext.cs:line 240
   at Microsoft.Bot.Builder.Solutions.Dialogs.RouterDialog.OnContinueDialogAsync(DialogContext innerDc, CancellationToken cancellationToken) in D:\a\1\s\sdk\csharp\libraries\microsoft.bot.builder.solutions\Dialogs\RouterDialog.cs:line 55
   at Microsoft.Bot.Builder.Dialogs.ComponentDialog.ContinueDialogAsync(DialogContext outerDc, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\ComponentDialog.cs:line 119
   at Microsoft.Bot.Builder.Dialogs.DialogContext.ContinueDialogAsync(CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\DialogContext.cs:line 240
   at Microsoft.Bot.Builder.Dialogs.DialogExtensions.RunAsync(Dialog dialog, ITurnContext turnContext, IStatePropertyAccessor`1 accessor, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\DialogExtensions.cs:line 32
   at CalendarSkill.Bots.DialogBot`1.OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken) in C:\Users\lamil\source\repos\botframework-solutions\skills\csharp\calendarskill\Bots\DialogBot.cs:line 39
   at CalendarSkill.Test.Flow.CalendarSkillTestBase.<>c__DisplayClass14_0.<<GetTestFlow>b__0>d.MoveNext() in C:\Users\lamil\source\repos\botframework-solutions\skills\csharp\tests\calendarskill.tests\Flow\CalendarSkillTestBase.cs:line 120
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.Solutions.Middleware.EventDebuggerMiddleware.OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken) in D:\a\1\s\sdk\csharp\libraries\microsoft.bot.builder.solutions\Middleware\EventDebuggerMiddleware.cs:line 60
   at Microsoft.Bot.Builder.MiddlewareSet.ReceiveActivityWithStatusAsync(ITurnContext turnContext, BotCallbackHandler callback, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder\MiddlewareSet.cs:line 55
   at Microsoft.Bot.Builder.BotAdapter.RunPipelineAsync(ITurnContext turnContext, BotCallbackHandler callback, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder\BotAdapter.cs:line 182
   at Microsoft.Bot.Builder.BotAdapter.RunPipelineAsync(ITurnContext turnContext, BotCallbackHandler callback, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder\BotAdapter.cs:line 192
   at Microsoft.Bot.Builder.Adapters.TestAdapter.ProcessActivityAsync(Activity activity, BotCallbackHandler callback, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder\Adapters\TestAdapter.cs:line 156
   at Microsoft.Bot.Builder.Adapters.TestFlow.<>c__DisplayClass8_0.<<Send>b__0>d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\Adapters\TestFlow.cs:line 108
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.Adapters.TestFlow.<>c__DisplayClass17_0.<<AssertReply>b__0>d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\Adapters\TestFlow.cs:line 325
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.Adapters.TestFlow.<>c__DisplayClass8_0.<<Send>b__0>d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\Adapters\TestFlow.cs:line 108
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.Adapters.TestFlow.<>c__DisplayClass17_0.<<AssertReply>b__0>d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\Adapters\TestFlow.cs:line 325
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.Adapters.TestFlow.<>c__DisplayClass17_0.<<AssertReply>b__0>d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\Adapters\TestFlow.cs:line 325
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.Adapters.TestFlow.<>c__DisplayClass8_0.<<Send>b__0>d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\Adapters\TestFlow.cs:line 108
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.Adapters.TestFlow.<>c__DisplayClass17_0.<<AssertReply>b__0>d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\Adapters\TestFlow.cs:line 325
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.Adapters.TestFlow.<>c__DisplayClass17_0.<<AssertReply>b__0>d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\Adapters\TestFlow.cs:line 325
--- End of stack trace from previous location where exception was thrown ---
   at CalendarSkill.Test.Flow.CreateCalendarFlowTests.Test_CalendarCreate_ConfirmNo_ChangeContent() in C:\Users\lamil\source\repos\botframework-solutions\skills\csharp\tests\calendarskill.tests\Flow\CreateCalendarFlowTests.cs:line 274
Result Message:	
Test method CalendarSkill.Test.Flow.CreateCalendarFlowTests.Test_CalendarCreate_ConfirmNo_ChangeContent threw exception: 
System.InvalidCastException: Unable to cast object of type 'System.Int64' to type 'Newtonsoft.Json.Linq.JToken'.

```

### CODE REVIEW

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.5.3)

**Comment by lauren-mills** [comment]
@cleemullins I added the datetime test, but stylecop won't let me add the Int64 test explicitly 😕

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.5.3)

File: libraries/Microsoft.Bot.Builder.Dialogs/ObjectPath.cs [hunk 1]
```diff
@@ -447,15 +447,16 @@ private static object GetNormalizedValue(object value, bool json)
...
+                       value is short || value is int || value is long ||
+                       value is ushort || value is uint || value is ulong ||
+                       value is decimal || value is float || value is double)
                 {
                     val = JValue.FromObject(value);
                 }
                 else
                 {
-                    val = (JToken)JsonConvert.DeserializeObject(JsonConvert.SerializeObject(value, expressionCaseSettings));
+                    val = JsonConvert.DeserializeObject(JsonConvert.SerializeObject(value, expressionCaseSettings));
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs/ObjectPath.cs]: shouldn't have been casting to JToken

File: libraries/Microsoft.Bot.Builder.Dialogs/ObjectPath.cs [hunk 2]
```diff
@@ -447,15 +447,16 @@ private static object GetNormalizedValue(object value, bool json)
                     val = null;
                 }
                 else if (value is string || value is byte || value is bool ||
-                        value is short || value is int || value is long ||
-                        value is ushort || value is uint || value is ulong ||
-                        value is decimal || value is float || value is double)
+                       value is DateTime || value is DateTimeOffset ||
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs/ObjectPath.cs]: >value is DateTime || value is DateTimeOffset || [](start = 23, length = 47)

this is the more correct solution

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.5.3)

**Pull request merged**